usage/src/main/java/com/cloud/usage/parser/BackupUsageParser.java

```diff
@@ -68,18 +68,7 @@ public static boolean parse(AccountVO account, Date startDate, Date endDate) {
             return true;
         }
 
-        final Map<Long, BackupInfo> vmUsageMap = new HashMap<>();
-        for (final UsageBackupVO usageBackup : usageBackups) {
-            final Long vmId = usageBackup.getVmId();
-            final Long zoneId = usageBackup.getZoneId();
-            final Long offeringId = usageBackup.getBackupOfferingId();
-            if (vmUsageMap.get(vmId) == null) {
-                vmUsageMap.put(vmId, new BackupUsageParser.BackupInfo(new Backup.Metric(0L, 0L), zoneId, vmId, offeringId));
-            }
-            final Backup.Metric metric = vmUsageMap.get(vmId).getMetric();
-            metric.setBackupSize(metric.getBackupSize() + usageBackup.getSize());
-            metric.setDataSize(metric.getDataSize() + usageBackup.getProtectedSize());
-        }
+        Map<String, BackupInfo> vmUsageMap = getVmUsageMap(usageBackups);
```

Comment:
@weizhouapache should we handle it by ensure everytime the offering is changed, a usage event is handled? How does other resources say VM handle this when their offerings are changed, maybe we should follow the same logic? (I've not checked, so maybe your approach is in-line with that)

Reply:
@rhtyd 
updated this pr to display the duration of backup offering in the period (see Pearl's testing result).
since you are the author of backup and restore framework, could you please review it ?